### DISCUSSION TITLE
Replacing dependency for RHEL8, netcat -> nmap-ncat

### DISCUSSION THREAD

**Discussion by ReifiedException** [netdata/netdata#12357] [Ideas]
### Reasons for replacement:
`netcat` is the only package we install from EPEL (on this platform). It also has two libraries as dependencies, both from EPEL, too: `libbsd` and `libretls`. `nmap-ncat` has no dependencies. In this way we could exclude EPEL from netdata agent requirements.

**Comment by Ferroin** [comment]
Already mentioned on Slack, but I’m of the opinion that we should take the time to evaluate whether we even need netcat at all.

At minimum, we probably should make this a weak dependency, and better than just that we should probably depend on the binary we need, not the package (for example `Requires: /usr/bin/nc` instead of `Requires: netcat`), as that will allow greater flexibility in what can satisfy the dependency.

**Reply by Ferroin** [reply]
OK, I’ve looked into our current code, and we currently use netcat (as `nc`) for the following things:

* IRC notification support in `health/notifications/alarm-notify.sh.in`
* As part of an exporting proxy in `exporting/nc-exporting.sh`
* An equivalent to the above exporting thing in the backends code (slated for removal before the next release, so we can ignore this one).

Given the relatively niche nature of the above usage, I think we’re safe converting these to soft dependencies that are not installed by default (`Suggests` in DEB/RPM terms) and updating things to depend not on the package, but on the command itself in the RPM packages (and seeing if there is a virtual package we can depend on in the DEB packages).

I’ll look at putting together a PR to do this later today.
